tidy3d/components/heat_charge/monitor_data.py

```diff
@@ -235,4 +242,220 @@ def symmetry_expanded_copy(self) -> VoltageData:
+        None, title="Data series", description="Contains the data."
+    )
+
+    capacitance_curve: Optional[Dict[str, CapacitanceCurveDataArray]] = pd.Field(
+        None,
+        title="Capacitance curve",
+        description="Small signal capacitance associated to the monitor.",
+    )
+
+    @pd.validator("data_series", always=True)
+    @skip_if_fields_missing(["monitor"])
+    def warn_no_data(cls, val, values):
+        """Warn if no data provided."""
+
+        mnt = values.get("monitor")
+
+        if val is None:
+            log.warning(
+                f"No data is available for monitor '{mnt.name}'. This is typically caused by "
+                "monitor not intersecting any solid medium."
+            )
+
+        return val
+
+    @property
+    def symmetry_expanded_copy(self) -> ChargeSimulationData:
+        """Return copy of self with symmetry applied."""
+
+        new_series = {}
+
+        for key, data in self.data_series.field_series.items():
+            new_series[key] = self._symmetry_expanded_copy(property=data)
+
+        return self.updated_copy(data_series=self.data_series.updated_copy(field_series=new_series))
+
+    def field_name(self, val: str) -> str:
+        """Gets the name of the fields to be plot."""
+        if val == "abs^2":
+            return "|V|², Electrons², Holes², Donors², Acceptors²"
+        else:
+            return "V, Electrons, Holes, Donors, Acceptors"
+
+
+class TemporalHeatChargeDataset(Tidy3dBaseModel):
+    """Class to deal with time-varying device fields."""
+
+    base_data: FieldDataset = pd.Field(title="Base data", description="Spatial dataset")
+
+    field_time_series: Tuple[IndexedDataArray, ...] = pd.Field(
+        title="Field time-series", description="Field values at different time steps."
+    )
+
+    time_steps_array: Tuple[pd.NonNegativeInt, ...] = pd.Field(
+        title="Time stes array",
+        description="Array containing the time steps at which the field is stored.",
+    )
+
+    @cached_property
+    def num_time_steps(self):
+        """Number of time steps in series"""
+        return len(self.field_time_series)
+
+    def return_time_step(self, time_step: int):
+        """Add a little description"""
+
+        assert time_step < self.num_time_steps
+        return self.base_data.updated_copy(values=self.field_time_series[time_step])
+
+    @pd.root_validator(skip_on_failure=True)
+    def check_data_and_times_have_same_length(cls, values):
+        """Check that both time series and data series have the same length."""
+
+        field_time_series = values["field_time_series"]
+        time_steps_array = values["time_steps_array"]
+        assert len(field_time_series) == len(time_steps_array)
+
+        return values
+
+
+class TemporalData(HeatChargeMonitorData):
+    """Data associated with a :class:`TemporalVoltageMonitor`: spatial voltage field.
+
+    Example
+    -------
+    >>> import tidy3d as td
```

Comment:
Wonder if we need a more abstract class at a higher level on the same lines of `FieldTimeData`. Ultimately compatibility now on this will save us a lot of pain in the future. Just wonder best implementation strategy